dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavTool.java

```diff
@@ -0,0 +1,301 @@
+    @Override
+    public void init(Object initData) {
+        context = (ViewContext) initData;
+        try {
+            this.request = context.getRequest();
+            this.currenthost = WebAPILocator.getHostWebAPI()
+                .getCurrentHost(context.getRequest());
+            this.currentLanguage = WebAPILocator.getLanguageWebAPI()
+                .getLanguage(this.request)
+                .getId();
+        } catch (Exception e) {
+            Logger.warn(this, e.getMessage(), e);
+        }
+    }
+
+    protected NavResult getNav(Host host, String path) throws DotDataException, DotSecurityException {
+        return getNav(host, path, this.currentLanguage, this.systemUser);
+    }
+
+    protected NavResultHydrated getNav(Host host, String path, long languageId, User systemUserParam)
+            throws DotDataException, DotSecurityException {
+
+        if (path != null && path.contains(".")) {
+            path = path.substring(0, path.lastIndexOf("/"));
+        }
+
+        Folder folder = !path.equals("/") ? APILocator.getFolderAPI()
+            .findFolderByPath(path, host, systemUserParam, true)
+                : APILocator.getFolderAPI()
+                    .findSystemFolder();
+        if (folder == null || !UtilMethods.isSet(folder.getIdentifier())) {
+            return null;
+        }
+        NavResult result = CacheLocator.getNavToolCache()
+            .getNav(host.getIdentifier(), folder.getInode(), languageId);
+        if (result != null) {
+
+            return new NavResultHydrated(result, this.context);
+        } else {
+            String parentId;
+            if (!folder.getInode()
+                .equals(FolderAPI.SYSTEM_FOLDER)) {
+                Identifier ident = APILocator.getIdentifierAPI()
+                    .find(folder);
+                parentId = ident.getParentPath()
+                    .equals("/") ? FolderAPI.SYSTEM_FOLDER
+                            : APILocator.getFolderAPI()
+                                .findFolderByPath(ident.getParentPath(), host, systemUserParam, false)
+                                .getInode();
+            } else {
+                parentId = null;
+            }
+            result = new NavResult(parentId, host.getIdentifier(), folder.getInode(), languageId);
+
+            Identifier ident = APILocator.getIdentifierAPI()
+                .find(folder);
+            result.setHref(ident.getURI());
+            result.setTitle(folder.getTitle());
+            result.setOrder(folder.getSortOrder());
+            result.setType("folder");
+            result.setPermissionId(folder.getPermissionId());
+            List<NavResult> children = new ArrayList<NavResult>();
+            List<String> folderIds = new ArrayList<String>();
+            result.setChildren(children);
+            result.setChildrenFolderIds(folderIds);
+            result.setShowOnMenu(folder.isShowOnMenu());
+
+            List menuItems;
+            if (path.equals("/"))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the "/" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1132)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Missing curly brace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00121)